openmc/tallies.py

```diff
@@ -452,9 +449,13 @@ def add_score(self, score):
                   'not a string'.format(score, self.id)
             raise ValueError(msg)
 
-        # If the score is already in the Tally, don't add it again
+        # If the score is already in the Tally, raise an error
         if score in self.scores:
-            return
+            msg = 'Unable to add a duplicate score {0} to Tally ID="{1}" ' \
```

Comment:
I think the long term solution is just to remove duplicate scores in `input_xml.F90`. I would offer to do that in this PR, but I looked at `input_xml.F90` and I could not find a simple solution to remove tallies so I would first ask if someone more familiar with `input_xml.F90` would be able to make a PR for the issue.

As an example of the current behavior of the Python API with duplicate scores, let's say we create the following tally:

    tally = openmc.Tally(name='n2n')
    tally.add_filter(energy_filter)
    tally.add_score('n2n')                # tally n2n
    tally.add_score('flux')               # tally flux
    tally.add_score('16')                 # tally 16 (which is the reaction for n2n)
    tallies_file.add_tally(tally)

When we read in the tally and print out the pandas data frame we get:

    n2n = sp.get_tally(name='n2n')
    n2n.get_pandas_dataframe()

![python_api_duplicate_tallies](https://cloud.githubusercontent.com/assets/1244537/12004353/37138df6-ab08-11e5-8865-a4d4dd707b64.png)

So I tend to view this "asymmetric" functionality as a halfway solution to removing duplicate scores. If we want a full solution to remove the asymmetry, perhaps someone (it might have to be me) could remove duplicate scores from `input_xml.F90`?